src/routes/queue.routes.js

```diff
@@ -1,57 +1,108 @@
 const express = require('express')
 const queue_routes = express.Router()
-const Redis = require('ioredis')
-const redis = new Redis()
 
-const queue = (client) => {
-  return{
+const healthyQueue = require('../queue/patient-queue')
 
-    push: async (key,val) => await client.rpush(key,val),
-    pop: async (key) => await client.lpop(key),
-    range: async (key) => await client.lrange(key,0,-1)
+queue_routes.get('/queue', async(req, res) => {
+  res.status(200).send(healthyQueue)
+})
+
+queue_routes.post('/queue/enqueue', async (req, res) => {
+
+  let priority = req.body.queue
+  let patient = req.body.patient
+
+  healthyQueue[priority].enqueue(patient)
+
+  // emit an event
+  res.status(200).send(healthyQueue)
+})
+
+queue_routes.post('/queue/dequeue', async (req, res) => {
+
+  let queue = req.body.queue
+  let dequeuedPatient = healthyQueue[queue].dequeue()
+
+  // here add dequeued patient to an ER board (in database)
+  // let acceptedPatient = erBoard.create(dequeuedPatient)
+
+  res.status(200).send(healthyQueue)
+})
+
+queue_routes.get('/queue/redenqueue', async (req, res) => {
+
+  red.enqueue({ name: 'Ayrat', age: Math.floor(Math.random() * 100) })
+  res.status(200).send(red)
+})
+
+queue_routes.get('/queue/reddequeue', async (req, res) => {
+  red.dequeue()
+  res.status(200).send(red)
+})
+
+queue_routes.get('/queue/yellowenqueue', async (req, res) => {
+
+  yellow.enqueue({ name: 'Lorenzo', age: Math.floor(Math.random() * 100) })
+  res.status(200).send(yellow)
+})
+
+queue_routes.get('/queue/yellowdequeue', async (req, res) => {
+
+  yellow.dequeue()
+  res.status(200).send(yellow)
+})
+
+queue_routes.get('/queue/redtoyellow', async (req, res) => {
+
+  let redPatient = red.dequeue()
+  if (redPatient) {
+    yellow.enqueue(redPatient)
   }
-}
+  res.status(200).send(yellow)
+})
+
+//==========================================================================
 
-queue_routes.get('/queue/all', async (req,res)=>{
+queue_routes.get('/queue/all', async (req, res) => {
   const { range } = queue(redis)
-  try{
-    const all_res = await Promise.all([range('red'),range('yellow'),range('green')]).then(res => res).catch(e => e)
```

Comment:
Think these Promises are older redis things. The queue class should now be synchronous, so no need to await interaction with it.